src/IECoreMaya/SceneShapeSubSceneOverride.cpp

```diff
@@ -0,0 +1,1615 @@
+	ensureFaceVaryingData( indicesTriangulated, p, n, wireframeIndicesReadable, linearInterpolation );
+
+	geometryData->positionData = p;
+	geometryData->normalData = n;
+	geometryData->uvData = uv;
+
+	// Prepare the index buffer for rendering a solid mesh
+	geometryData->indexData = indicesTriangulated;
+
+	IECore::IntVectorDataPtr wireframeIndexData( new IntVectorData() );
+	std::vector<int> &wireframeIndexReadable = wireframeIndexData->writable();
+
+	const IECore::IntVectorData *numVerticesPerFace = meshPrimitive->verticesPerFace();
+	if( numVerticesPerFace )
+	{
+		const std::vector<int> &numVerticesPerFaceReadable = numVerticesPerFace->readable();
+
+		int totalNumEdgeIndices = std::accumulate( numVerticesPerFaceReadable.begin(), numVerticesPerFaceReadable.end(), 0 ) * 2;
+		wireframeIndexReadable.reserve( totalNumEdgeIndices );
+
+		int offset = 0;
+		for( int numVerts : numVerticesPerFaceReadable )
+		{
+			for( int k = 0; k < numVerts; ++k )
+			{
+				wireframeIndexReadable.push_back( wireframeIndicesReadable[offset + k] );
+				wireframeIndexReadable.push_back( wireframeIndicesReadable[offset + (k+1)%numVerts] );
+			}
+			offset += numVerts;
+		}
+	}
+	geometryData->wireframeIndexData = wireframeIndexData;
+}
+
+GeometryDataPtr geometryGetter( const GeometryDataCacheGetterKey &key, size_t &cost )
+{
+	GeometryDataPtr geometryData( new GeometryData() );
+
+	if( !key.object )
+	{
+		fillBoundData( key.bounds, geometryData );
+		return geometryData;
+	}
+
+	switch( static_cast<IECoreScene::TypeId>( key.object->typeId() ) )
+	{
+	case IECoreScene::TypeId::MeshPrimitiveTypeId :
+		fillMeshData( key.object, key.descriptorList, geometryData ); break;
+	case IECoreScene::TypeId::PointsPrimitiveTypeId :
+		fillPointsData( key.object, geometryData ); break;
+	case IECoreScene::TypeId::CurvesPrimitiveTypeId :
+		fillCurvesData( key.object, geometryData ); break;
+	default :
+		// we should never get here as only the above types are supported, see `objectCanBeRendered`.
+		break;
+	}
+
+	// \todo compute cost from geometry data
+	cost = 1;
+
+	return geometryData;
+}
+
+// Cache for geometric data
+using GeometryDataCache = IECore::LRUCache<IECore::MurmurHash, GeometryDataPtr, IECore::LRUCachePolicy::Parallel, GeometryDataCacheGetterKey>;
+GeometryDataCache g_geometryDataCache(geometryGetter, 100);
+
+MString renderItemName( const std::string &baseName, RenderStyle style )
+{
+	switch( style )
+	{
+	case RenderStyle::BoundingBox :
+		return MString( (baseName + "_bb").c_str() );
+	case RenderStyle::Wireframe :
+		return MString( (baseName + "_wf").c_str() );
+	case RenderStyle::Solid :
+		return MString( (baseName + "_s").c_str() );
+	case RenderStyle::Textured :
+		return MString( (baseName + "_t").c_str() );
+	default :
+		return MString();
+	}
+}
+
+MRenderItem *getRenderItem( MSubSceneContainer &container, ConstObjectPtr object, const MString &name, RenderStyle style, bool &isEmpty )
```

Comment:
> Perhaps createOrAcquireRenderItem...

For what it's worth, in Gaffer `create()` methods always return a new instance, and `acquire()` methods either create one if it doesn't exist or reuse one that exists already. So in Gaffer parlance this would just be `acquireRenderItem()`...

Reply:
Addressed in cdc2c2a. Using gaffer lingo now.